xbmc/interfaces/AnnouncementManager.cpp

```diff
@@ -248,5 +276,29 @@ void CAnnouncementManager::Announce(AnnouncementFlag flag, const char *sender, c
   if (id > 0)
     object["item"]["id"] = id;
 
-  Announce(flag, sender, message, object);
+  DoAnnounce(flag, sender, message, object);
+}
+
+void CAnnouncementManager::Process()
+{
+  SetPriority(GetMinPriority());
+
+  while (!m_bStop)
+  {
+    CSingleLock lock (m_critSection);
+    if (!m_announcementQueue.empty())
+    {
+      auto announcement = m_announcementQueue.front();
+      m_announcementQueue.pop_front();
+      {
+        CSingleExit ex(m_critSection);
+        DoAnnounce(announcement.flag, announcement.sender.c_str(), announcement.message.c_str(), announcement.item, announcement.data);
+      }
+    }
+    else
+    {
+      CSingleExit ex(m_critSection);
+      m_queueEvent.WaitMSec(1000);
```

Comment:
> Just change 1000 to infinite an see. If it works, we're good, otherwise this issue will be clear.

Ok, I'll will take this as an offer that you'll observe this. Please don't get me wrong, I don't fight to keep my changes here, actually I'd prefer not even be active in this area :)
In the real-time space I mostly take care it is not uncommon that you miss a an event. Not really the event itself but the event in a particular state. Waking up after a certain period of time keeps the system alive.